dist/cas-auth-api.js

```diff
@@ -1,285 +1,449 @@
+                //Exchange ticket for access_token
+                $injector
+                    .get('$http')
+                    .get(apiUrl('token'), {
+                        params: {
+                            st: st
+                        }
+                    })
+                    .then(tokenResponder, tokenResponderError);
+            }
+
+            function ticketResponderError() {
+                standardAuthenticationError('ERR_TICKET', 'Failed to fetch ticket.');
+            }
+
+            function serviceResponder(serviceResponse) {
+                var url, httpConfig;
+                if (config.oAuth === true) {
+                    // Use casTicketPath to fetch ticket for given URL
+                    url = config.casBaseUrl + config.casTicketPath;
+                    httpConfig = {
+                        params: {
+                            service: serviceResponse.data.data.id
+                        },
+                        headers: {
+                            'Authorization': 'Bearer ' + getAccessTokenFromHash().access_token
+                        }
+                    };
+                } else {
+                    // Use wrapper ticketUrl to fetch ticket for given URL
+                    url = config.ticketUrl;
+                    httpConfig = {
+                        params: {
+                            service: serviceResponse.data.data.id
+                        }
+                    };
+                }
+                $injector.get('$http').get(url, httpConfig).then(ticketResponder, ticketResponderError);
+            }
+
+            function serviceResponderError() {
+                standardAuthenticationError('ERR_SERVICE', 'Failed to fetch service.');
+            }
+
             /**
-             * Begins the authentication process
-             */
+            * Begins the authentication process
+            */
             function beginAuthentication() {
                 // Requests already deferred, bail.
-                if (angular.isObject(_deferredAuthentication)) {
+                if (angular.isObject(config.deferredAuthentication)) {
                     return;
                 }
 
                 // Create new deferred to handle authentication process
-                _deferredAuthentication = $q.defer();
+                config.deferredAuthentication = $q.defer();
 
                 // invalidate access token and cache
-                _accessToken = undefined;
-                if (_cacheAccessToken) {
+                config.accessToken = undefined;
+                if (config.cacheAccessToken && angular.isDefined(lscache)) {
                     lscache.remove('access_token');
                 }
 
+                if (config.oAuth === true && getAccessTokenFromHash().access_token === undefined) {
+                    $injector.get('$window').location.href =
```

Comment:
Does this handle encoding any invalid characters in the url, if say clientId or redirectUrl have characters that need to be escaped?